frontend/components/ChatBox.tsx

```diff
@@ -0,0 +1,88 @@
+'use client';
+
+import React, { useState, useEffect, useRef } from 'react';
+import { Text, Stack, Input, Button, Flex } from '@chakra-ui/react';
+import useAuth from '@/hooks/useAuth';
+import { io, Socket } from 'socket.io-client';
+
+interface ChatBoxProps {
+	users: string[];
+	roomId: string; 
+}
+
+interface Message {
+	senderUsername: string;
+	text: string;
+}
+
+export const ChatBox = ({ users, roomId }: ChatBoxProps) => {
+	const username = useAuth().username;
+	const [messages, setMessages] = useState<Message[]>([]);
+	const [newMessage, setNewMessage] = useState('');
+	const messagesEndRef = useRef<HTMLDivElement>(null);
+	const socketRef = useRef<Socket | null>(null);
+
+	useEffect(() => {
+		const newSocket = io('http://localhost:8009');
```

Comment:
Might want to use a env variable for this

Reply:
Or actually nvm, I can update this part when configuring for cloud deployment altogether